test-d/require-one.ts

```diff
@@ -0,0 +1,24 @@
+import {expectType, expectError} from 'tsd';
+import {RequireOne} from '..';
+
+type SystemMessages = {
+	default: string;
+
+	macos: string;
+	linux: string;
+
+	optional?: string;
+};
+
+type ValidMessages = RequireOne<SystemMessages, 'macos' | 'linux'>;
+const test = (_: ValidMessages): void => {};
+
+test({macos: 'hey', default: 'hello'});
+test({linux: 'sup', optional: 'howdy', default: 'hello'});
+
+expectError(test({}));
+expectError(test({macos: 'hey', linux: 'sup', default: 'hello'}));
```

Comment:
This is quite unexpected. So if I understand it correctly, the properties `macos` and `linux` are mutually exclusive in this type? Is this intended? Intuitively, I would expect that one of the both is required but that it's OK if I'd provide both.

Any ideas?

Reply:
Yup, it is intended, this is how it differs from `RequireAtLeastOne`. Still not sure about the naming, it was somewhat discussed in the previous stale pull request, but I don't think any consensus was found.

Reply:
What do you think, @sindresorhus?

Reply:
I agree, `RequireOne` is not clear enough about this. `RequireExactlyOne` sounds good.